app-web/src/components/Form/TopicForm.js

```diff
@@ -140,31 +177,34 @@ export const TopicForm = () => {
   );
 };
 
-const convertToRegistryFormat = values => {
+const convertToRegistryFormat = (values, operation) => {
   const convertedFields = {
     name: values.topicName,
     description: values.topicDescription,
     sourceProperties: {
       sources: values.sources.map(source => ({
         sourceType: source.sourceType,
-        sourceProperties: getSourceProps(source),
+        sourceProperties: getSourceProps(source, operation),
         resourceType: source.resourceType,
       })),
     },
   };
   return convertedFields;
 };
 
-const getSourceProps = source => {
+const getSourceProps = (source, operation) => {
   const properties = {};
-  properties.url = source.url;
+  properties.url = source.sourceProperties.url;
   if (source.sourceType === 'github') {
-    properties.owner = source.owner;
-    properties.repo = source.repo;
-    properties.files = [source.file];
+    properties.owner = source.sourceProperties.owner;
+    properties.repo = source.sourceProperties.repo;
+    properties.files =
+      operation === 'create'
+        ? [source.sourceProperties.files.split(',')]
+        : String(source.sourceProperties.files).split(',');
```

Comment:
This seems funny... You typically do not use the String constructor without `new`